lib/ansible/utils/_ssh_agent.py

```diff
@@ -0,0 +1,674 @@
+
+
+class SshAgentClient:
+    def __init__(self, auth_sock: str) -> None:
+        self._sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        self._sock.settimeout(_SSH_AGENT_CLIENT_SOCKET_TIMEOUT)
+        self._sock.connect(auth_sock)
+
+    def close(self) -> None:
+        self._sock.close()
+
+    def __enter__(self) -> t.Self:
+        return self
+
+    def __exit__(
+            self,
+            exc_type: type[BaseException] | None,
+            exc_value: BaseException | None,
+            traceback: types.TracebackType | None
+    ) -> None:
+        self.close()
+
+    def send(self, msg: bytes) -> bytes:
+        length = uint32(len(msg)).to_blob()
+        self._sock.sendall(length + msg)
+        bufsize = uint32.from_blob(self._sock.recv(4))
+        resp = self._sock.recv(bufsize)
+        if resp[0] == ProtocolMsgNumbers.SSH_AGENT_FAILURE:
+            raise SshAgentFailure('agent: failure')
+        return resp
+
+    def remove_all(self) -> None:
+        self.send(
+            ProtocolMsgNumbers.SSH_AGENTC_REMOVE_ALL_IDENTITIES.to_blob()
+        )
+
+    def remove(self, public_key: CryptoPublicKey) -> None:
+        key_blob = PublicKeyMsg.from_public_key(public_key).to_blob()
+        self.send(
+            ProtocolMsgNumbers.SSH_AGENTC_REMOVE_IDENTITY.to_blob() +
+            uint32(len(key_blob)).to_blob() + key_blob
+        )
+
+    def add(
+            self,
+            private_key: CryptoPrivateKey,
+            comments: str | None = None,
+            lifetime: int | None = None,
+            confirm: bool | None = None,
+    ) -> None:
+        key_msg = PrivateKeyMsg.from_private_key(private_key)
+        key_msg.comments = unicode_string(comments or '')
+        if lifetime:
+            key_msg.constraints += constraints(
+                [ProtocolMsgNumbers.SSH_AGENT_CONSTRAIN_LIFETIME]
+            ).to_blob() + uint32(lifetime).to_blob()
+        if confirm:
+            key_msg.constraints += constraints(
+                [ProtocolMsgNumbers.SSH_AGENT_CONSTRAIN_CONFIRM]
+            ).to_blob()
+
+        if key_msg.constraints:
+            msg = ProtocolMsgNumbers.SSH_AGENTC_ADD_ID_CONSTRAINED.to_blob()
+        else:
+            msg = ProtocolMsgNumbers.SSH_AGENTC_ADD_IDENTITY.to_blob()
+        msg += key_msg.to_blob()
+        self.send(msg)
+
+    def list(self) -> KeyList:
+        req = ProtocolMsgNumbers.SSH_AGENTC_REQUEST_IDENTITIES.to_blob()
+        r = memoryview(bytearray(self.send(req)))
+        if r[0] != ProtocolMsgNumbers.SSH_AGENT_IDENTITIES_ANSWER:
+            raise SshAgentFailure(
+                'agent: non-identities answer received for identities list'
+            )
+        return KeyList.from_blob(r[1:])
+
+    def __contains__(self, public_key: CryptoPublicKey) -> bool:
+        msg = PublicKeyMsg.from_public_key(public_key)
+        return msg in self.list().keys
+
+
+def load_private_key(key_data: bytes, passphrase: bytes) -> CryptoPrivateKey:
```

Comment:
The connection plugin passes `to_bytes(passphrase) if passphrase else None`:
```suggestion
def load_private_key(key_data: bytes, passphrase: bytes | None) -> CryptoPrivateKey:
```